### DISCUSSION TITLE
the role entity in the grant entity is invalid

### DISCUSSION THREAD

**Discussion by yuyushao** [milvus-io/milvus#35277] [Q&A and General discussion]
已有用户，之前登录正常，突然登录开始报错：
[2024/08/05 07:24:40.986 +00:00] [INFO] [rootcoord/root_coord.go:950] ["failed to list databases"] [traceID=036a37b004e8a9f6171f9276d1c02ba5] [msgID=451632871754170374] [error="the role entity in the grant entity is invalid"]

尝试了以下方式重建授权：
1、user 和角色名 及权限完全一致 -- 失败，报一样的错误
2、user 和权限一致，角色名改名 -- 失败，报一样的错误
3、user改名，角色名 及权限完全一致 -- 成功。

请问可能什么问题导致的？谢谢~

**Comment by yhmo** [comment]
Wait @SimFG to comment.

**Comment by SimFG** [comment]
You can try to output all the role names. From the error message, it seems that the role name is not valid.

**Reply by xiaofan-luan** [reply]
1. any operation you do?
2. can you do list user and list role and show the result?
3. what is the code user access  milvus?

**Reply by yuyushao** [reply]
1、upgrade 2.4.0 to 2.4.5 ,and restart 
2、yes
3、attu
